packages/m2m-gateway/src/services/agreementService.ts

```diff
@@ -414,18 +414,27 @@ export function agreementServiceBuilder(
         path: storagePath,
       };
 
-      const { data: document, metadata } =
-        await clients.agreementProcessClient.addAgreementConsumerDocument(
-          documentSeed,
-          {
-            params: { agreementId },
-            headers,
-          }
+      try {
+        const { data: document, metadata } =
+          await clients.agreementProcessClient.addAgreementConsumerDocument(
+            documentSeed,
+            {
+              params: { agreementId },
+              headers,
+            }
+          );
+
+        await pollAgreementById(agreementId, metadata, headers);
```

Comment:
consider moving the polling outside try-catch.

Edge case:
1. `addAgreementConsumerDocument` to `agreementProcess` succeeds
2. `agreement-readmodel-writer` is in error
3. polling fails
4. the document is deleted, but the event was successfully stored
5. `agreement-readmodel-writer` is fixed, the agreement contains a reference to a deleted document